### ISSUE TITLE
Using mask in the inpainting process, the mask is not encoded into latent space, but why?

### ISSUE THREAD

**Issue by himmetozcan** [CompVis/stable-diffusion#521]
I am trying to understand the usage of "mask" in the inpainting process. It looks like it is being used in these [lines](https://github1s.com/CompVis/stable-diffusion/blob/HEAD/scripts/inpaint.py#L75-L79):
```
# encode masked image and concat downsampled mask
c = model.cond_stage_model.encode(batch["masked_image"])
cc = torch.nn.functional.interpolate(batch["mask"],
                                     size=c.shape[-2:])
c = torch.cat((c, cc), dim=1)
```
it looks like the mask is used in 2 places, one is `c` --> [here](https://github1s.com/CompVis/stable-diffusion/blob/HEAD/scripts/inpaint.py#L24-L25):
`batch["masked_image"]= (1-mask)*image`

and in [here](https://github1s.com/CompVis/stable-diffusion/blob/HEAD/scripts/inpaint.py#L77-L78) `cc` --> `cc = torch.nn.functional.interpolate(batch["mask"], size=c.shape[-2:])`

So this means:

1.  in `c`, the input image is multiplied by mask in pixel space and then encoded to latent space.
2. in `cc`, only mask image from pixel space is downscaled to dimensions of latent vector (it is not encoded to latent vector)

Then they are concatenated:
`c = torch.cat((c, cc), dim=1)`

**So in summary my question is below.**
The mask image is first downscaled to the dimensions of the latent vector, then concatenated with the encoded image (the result of the image being multiplied by the mask in pixel space and then encoded to the latent space). So this means that the local positions in the mask image will be directly related to the local positions in the latent vector ? The spatial relationship is preserved, how?

**Comment by antonzub99** [comment]
I suggest reading the full paper (https://arxiv.org/abs/2112.10752) or section 3.1 at least.

tl;dr: the compression network is learned in such a way that latent vectors are perceptually similar to original images, so yeah, spatial relationship is somewhat preserved.
